sympy/core/tests/test_dsubs.py

```diff
@@ -0,0 +1,55 @@
+from sympy import (S, symbols, Eq, Function, Derivative, asin, sin, cos, tan, exp, log, sqrt)
+from sympy.abc import a, b, m, n, t, u, w, x, y, E
+
+f, g, h, F, psi, H = symbols('f g h F psi H', cls=Function)
+hbar = symbols("hbar")
+
+def test_dsubs():
+    # Order 1
+    eq = f(x).diff(x) - f(x)
+    # Issue 17950
+    assert eq.dsubs({x: exp(y)}) == -f(y) + exp(-y)*Derivative(f(y), y)
+    assert eq.dsubs({x: exp(y), f(x): tan(g(y))}) == (tan(g(y))**2 + 1)*exp(-y)*Derivative(g(y), y) - tan(g(y))
+
+    # Order 2
+    eq = f(x).diff(x, 2)
+    assert eq.dsubs({x: a*y}, [y]) == f(y).diff(y, 2)/a**2
+
+    assert eq.dsubs({x: u**3}).simplify() == (f(u).diff(u, 2)*u - 2*f(u).diff(u))/(9*u**5)
+
+    assert eq.dsubs({x: t**2, f(x): h(t)**3}).simplify().expand() == 3*h(t)**2*Derivative(h(t), (t, 2))/(4*t**2) + \
+        3*h(t)*Derivative(h(t), t)**2/(2*t**2) - 3*h(t)**2*Derivative(h(t), t)/(4*t**3)
+
+    assert eq.dsubs({x: asin(t**3)}).simplify().collect(f(t).diff(t)) == ((-t**6 - 2)*Derivative(f(t), t) + \
+        (-t**7 + t)*Derivative(f(t), (t, 2)))/(9*t**5)
```

Comment:
I think it's better not to use `simplify` in tests like this. Note least because it means we're not testing the output from `dsubs` directly.

We should choose to either:

1. Have `dsubs` apply some light-weight but commonly needed simplifications (with an option to disable them).
2. Clearly explain in the docstring what simplification functions are likely to be useful.

Either way we should test that actual output of `dsubs` rather than what happens after simplifying that output.

Reply:
Ok, maybe we can add kwargs to the function which could be passed into `simplify`. I will change the tests

Reply:
In 2 tests, I used `expand` as the output was too long without it. Is this fine, or should the tests be changed?

Reply:
That's fine if it's needed in a couple of cases. If that's a good way of simplify the output in some cases then it should be mentioned in the docstring.

Reply:
Yes, in general, expanding the output helps. I will add it to the docstring.